beacon_node/network/src/sync/block_lookups/common.rs

```diff
@@ -371,16 +371,21 @@ impl<L: Lookup, T: BeaconChainTypes> RequestState<L, T> for BlobRequestState<L,
 
     fn verify_response_inner(
         &mut self,
-        _expected_block_root: Hash256,
+        expected_block_root: Hash256,
         blob: Option<Self::ResponseType>,
         peer_id: PeerId,
     ) -> Result<Option<FixedBlobSidecarList<T::EthSpec>>, LookupVerifyError> {
         match blob {
             Some(blob) => {
                 let received_id = blob.id();
                 if !self.requested_ids.contains(&received_id) {
+                    if expected_block_root == received_id.block_root {
+                        // We may have already received this blob via gossip since the lookup,
+                        // so don't return an error here.
+                        return Ok(None);
+                    }
```

Comment:
```suggestion
```
I think the issue is really that we update `requested_ids` mid-response which keeps us from tracking what we should receive. If this is the case, this is more treating a symptom than the cause. This is an alternative:

https://github.com/sigp/lighthouse/pull/5484/commits/77c4ffe44a8725ab23db8fe9310b98b3804b9df1#diff-af9c70c212795da7557f17c24d6fc12fa4aa6833d06db6eb2d807c277ee2aadbR260